pkg/http/delayed_writer.go

```diff
@@ -0,0 +1,67 @@
+package http
+
+import (
+	"net/http"
+)
+
+var (
+	_ http.Flusher        = (*DelayedWriter)(nil)
+	_ http.ResponseWriter = (*DelayedWriter)(nil)
+)
+
+// DelayedWriter is an implementation of http.ResponseWriter. It delays writing to the output until RealFlush is called.
+type DelayedWriter struct {
+	ResponseCode int
+
+	writer         http.ResponseWriter
+	headerRecorded bool
+
+	buffer        []byte
+	bytesBuffered int
+}
+
+// NewDelayedWriter creates an http.ResponseWriter that delays writing.
+func NewDelayedWriter(w http.ResponseWriter) *DelayedWriter {
+	return &DelayedWriter{
+		writer: w,
+	}
+}
+
+// Flush does nothing.
+func (dw *DelayedWriter) Flush() {}
```

Comment:
is it possible to do whatever `RealFlush` is doing here?

Reply:
Actually no... The point of having this delayed_writer is we can "hijack" the `Flush` function to let it do nothing. The `ServeHTTP` call will call `Flush` multiple times.

[This line](https://sourcegraph.com/github.com/knative/serving@f4ea3ac779621ea133a78a746525f6c6ca9947de/-/blob/cmd/queue/main.go?L231) initializes a `NewHeaderPruningReverseProxy`, which is a `reverseproxy`. And if you read the `ServeHTTP` code [here](https://go.dev/src/net/http/httputil/reverseproxy.go) (at line 214) you'll see how `Flush` is called inside it.